options_test.go

```diff
@@ -1,68 +1,82 @@
 package lightstep
 
 import (
-	"testing"
-
 	"github.com/lightstep/lightstep-tracer-go/basictracer"
-	"github.com/stretchr/testify/assert"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	opentracing "github.com/opentracing/opentracing-go"
 )
 
-func TestSpan_SetID3(t *testing.T) {
-	var (
+var _ = Describe("Options", func() {
+	const (
 		expectedTraceID      uint64 = 1
 		expectedSpanID       uint64 = 2
 		expectedParentSpanID uint64 = 3
 	)
 
-	recorder := basictracer.NewInMemoryRecorder()
-	tracer := basictracer.NewWithOptions(basictracer.Options{Recorder: recorder})
-
-	tracer.StartSpan("x", SetTraceID(1), SetSpanID(2), SetParentSpanID(3)).Finish()
-
-	spans := recorder.GetSpans()
-	assert.Equal(t, 1, len(spans))
-
-	span := spans[0]
-	assert.Equal(t, span.Context.TraceID, expectedTraceID)
-	assert.Equal(t, span.Context.SpanID, expectedSpanID)
-	assert.Equal(t, span.ParentSpanID, expectedParentSpanID)
-}
-
-func TestSpan_SetID2(t *testing.T) {
-	var (
-		expectedTraceID uint64 = 1
-		expectedSpanID  uint64 = 2
-	)
-
-	recorder := basictracer.NewInMemoryRecorder()
-	tracer := basictracer.NewWithOptions(basictracer.Options{Recorder: recorder})
-
-	tracer.StartSpan("x", SetTraceID(1), SetSpanID(2)).Finish()
-
-	spans := recorder.GetSpans()
-	assert.Equal(t, 1, len(spans))
-
-	span := spans[0]
-	assert.Equal(t, span.Context.TraceID, expectedTraceID)
-	assert.Equal(t, span.Context.SpanID, expectedSpanID)
-	assert.Equal(t, span.ParentSpanID, uint64(0))
-}
-
-func TestSpan_SetID1(t *testing.T) {
 	var (
-		expectedTraceID uint64 = 1
+		recorder *basictracer.InMemorySpanRecorder
+		tracer   opentracing.Tracer
 	)
 
-	recorder := basictracer.NewInMemoryRecorder()
-	tracer := basictracer.NewWithOptions(basictracer.Options{Recorder: recorder})
-
-	tracer.StartSpan("x", SetTraceID(1)).Finish()
-
-	spans := recorder.GetSpans()
-	assert.Equal(t, 1, len(spans))
-
-	span := spans[0]
-	assert.Equal(t, span.Context.TraceID, expectedTraceID)
-	assert.NotEqual(t, span.Context.SpanID, uint64(0))
-	assert.Equal(t, span.ParentSpanID, uint64(0))
-}
+	BeforeEach(func() {
+		recorder = basictracer.NewInMemoryRecorder()
+		tracer = basictracer.NewWithOptions(basictracer.Options{Recorder: recorder})
+	})
+
+	Describe("Categorizing setting trace options", func() {
+		Context("When only the TraceID is set", func() {
+			BeforeEach(func() {
+				tracer.StartSpan("x", SetTraceID(expectedTraceID)).Finish()
+			})
+
+			It("Should set the options appropriately", func() {
+				By("Only running one span")
```

Comment:
The second by block relies on there being at least one span returned from recorder.GetSpans(). If I separate them I will have to duplicate the check made by the first by block